src/main/java/nl/tudelft/ewi/devhub/server/database/controllers/Deliveries.java

```diff
@@ -94,6 +93,32 @@ public boolean lastDeliveryIsApprovedOrDisapproved(Assignment assignment, Group
             .sorted(byState.thenComparing(bySubmissionDate))
             .collect(Collectors.toList());
     }
+    
+    /**
+     * Get the most recent delivery for every group in this assignment
+     * @param assignment current assignment
+     * @return a list of deliveries
+     */
+    @Transactional
+    public List<Delivery> getLastDeliveries(Assignment assignment) {
+    	return this.getLastDeliveries(assignment, null);
+    }
+    
+    /**
+     * Get the most recent delivery for every group in this assignment with the specific state
+     * @param assignment current assignment
+     * @return a list of deliveries
+     */
+    @Transactional
+    public List<Delivery> getLastDeliveries(Assignment assignment, String state) {
```

Comment:
I have tried to use Delivery.State, but couldn't get the expression below working. Maybe I missed something.